### PR TITLE
Azurecompute ARM - proper tests for disks

### PR DESCRIPTION
**Pull request by bostko**
- Fix DataDisk.CachingTypes.fromValue to recognize values properly
- Fix tests which had blob disks and managed disks at the same time
- At this point managed disks are used by default and should be tested.

### CODE REVIEW

**Comment by bostko** [comment]
Examples if someone wants to add tests for managed disks operations https://docs.microsoft.com/en-us/rest/api/manageddisks/disks/disks-create-or-update

**Comment by bostko** [comment]
The problem which I was hitting without if I update disks section of a vm with :

```java
        DataDisk newDataDisk = DataDisk.builder().name("managed-disk")
//....
                .caching(DataDisk.CachingTypes.READ_WRITE) 
//.....
                .build();
```
and in request for updating disks of vm it was containing caching unrecognized which was making Azure ARM to return:
```
error": {
  "code": "BadRequest",
  "target": "vm.properties.storageProfile.dataDisks[0].caching",
  "message": "Error converting value \"Unrecognized\" to type 'System.Nullable`1[Microsoft.WindowsAzure.ComputeResourceProvider.API.Disk+CachingType]'. Path 'properties.storageProfile.dataDisks[0].caching', line 1, position 815."
 }
```

File: azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/domain/DataDisk.java [hunk 1]
```diff
@@ -108,13 +113,15 @@ public String toString() {
    public static DataDisk create(final String name, final String diskSizeGB, final Integer lun,
                                  final VHD vhd, final VHD image, final String createOption, final String caching, 
                                  final ManagedDiskParameters managedDiskParamenters, final String provisioningState) {
-      return builder()
-              .name(name)
+      final Builder builder = builder();
+      if (caching != null) {
```
Comment (bostko) [inline azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/domain/DataDisk.java]: Otherwise null will be treated as Unrecognized.

**Comment by nacx** [comment]
Pushed to [master](https://git1-us-west.apache.org/repos/asf?p=jclouds-labs.git;a=commit;h=b3da4b3) and [2.0.x](http://git-wip-us.apache.org/repos/asf/jclouds-labs/commit/2e41dc90). Thanks @bostko!

**Pull request closed without merging**